### PR TITLE
Add swagger files for api-version 2021-05-01-preview of MySQL flexible server

### PR DESCRIPTION
**Pull request by Suna-MS**
…e server

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [x] SDK of Java
      - [x] SDK of Js
      - [x] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-05-01-preview/mysql.json [hunk 1]
```diff
@@ -0,0 +1,2604 @@
...
+        "tier"
+      ],
+      "description": "Billing information related properties of a server."
+    },
+    "Network": {
+      "type": "object",
+      "properties": {
+        "publicNetworkAccess": {
+          "$ref": "#/definitions/EnableStatusEnum",
+          "readOnly": true,
```
Comment (pilor) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-05-01-preview/mysql.json:1595]: why is this readonly?  The description implies it is settable and the contract for publicNetworkAccess typically has this settable

Comment (Suna-MS) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-05-01-preview/mysql.json:1595]: In our design, the property of publicNetworkAccess is not settable, its value depends on if delegatedSubnetResourceId is provided or not when creating a server. If delegatedSubnetResourceId is provided in request body,  the value of publicNetworkAccess will be Disabled (server can only be accessed via vnet), else it is Enabled. 
	"network": {
            "delegatedSubnetResourceId": "xxxxx"
          }

File: specification/mysql/resource-manager/readme.md [hunk 2]
```diff
@@ -123,6 +123,15 @@ input-file:
 - Microsoft.DBforMySQL/preview/2020-07-01-preview/mysql.json
 ```
 
+### Tag: package-2021-05-01-preview
```
Comment (weidongxu-microsoft) [inline specification/mysql/resource-manager/readme.md:126]: similar to postresql, if you mean a separate SDK package, at least show it in tag name

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-05-01-preview/mysql.json (hunk 1 above)
Comment (pilor) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-05-01-preview/mysql.json:1595]: Please update the description of this property to reflect that

Comment (Suna-MS) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-05-01-preview/mysql.json:1595]: updated in new commit

File: specification/mysql/resource-manager/readme.md (hunk 2 above)
Comment (Suna-MS) [reply specification/mysql/resource-manager/readme.md]: updated the tag name to package-flexibleserver-2021-05-01-preview

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-05-01-preview/mysql.json [hunk 3]
```diff
@@ -0,0 +1,2606 @@
...
+        "properties": {
+          "$ref": "#/definitions/ServerProperties",
+          "x-ms-client-flatten": true,
+          "description": "Properties of the server."
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
```
Comment (weidongxu-microsoft) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-05-01-preview/mysql.json:1800]: `TrackedResource` already defined `tags`

**Review by weidongxu-microsoft — APPROVED** [review]

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-05-01-preview/mysql.json (hunk 3 above)
Comment (Suna-MS) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-05-01-preview/mysql.json]: thanks, removed tags in new commit.

**Comment by Suna-MS** [comment]
Hi, @weidongxu-microsoft , could you please help merge this pull request?

**Comment by weidongxu-microsoft** [comment]
Ignore tie Cross-Version Breaking Changes as this flexible server is totally different API than previous postgresql tag.

**Pull request merged**